client/src/components/ChatBox.jsx

```diff
@@ -0,0 +1,229 @@
+import ListItemAvatar from '@material-ui/core/ListItemAvatar';
+import ListItemText from '@material-ui/core/ListItemText';
+import { makeStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import classnames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import socketIOClient from 'socket.io-client';
+import { useVerify } from '../services/authenticationService';
+import { useGetConversationMessages, useSendConversationMessage } from '../services/chatService';
+
+const useStyles = makeStyles((theme) => ({
+	root: {
+		height: '100vh',
+		'& .MuiOutlinedInput-notchedOutline': {
+			borderColor: 'white'
+		}
+	},
+	headerRow: {
+		maxHeight: 60,
+		zIndex: 5
+	},
+	paper: {
+		display: 'flex',
+		alignItems: 'center',
+		height: '100%',
+		marginLeft: 3,
+		height: 90,
+		filter: 'drop-shadow(0px 2px 10px rgba(88,133,196,0.1))',
+		boxShadow: 'none'
+	},
+	messageContainer: {
+		display: 'flex',
+		alignContent: 'flex-end'
+	},
+	messagesRow: {
+		overflowY: 'auto'
+	},
+	newMessageRow: {
+		width: '100%',
+		padding: theme.spacing(0, 2, 1)
+	},
+	messageBubble: {
+		flex: '0 1 auto',
+		padding: 10,
+		border: '1px solid white',
+		color: theme.palette.primary.light,
+		backgroundImage: 'linear-gradient(225deg, #6cc1ff 0%, #3a8dff 100%)',
+		borderRadius: '0 10px 10px 10px',
+		marginTop: 8,
+		maxWidth: '40em',
+		color: 'white'
+	},
+	messageBubbleRight: {
+		borderRadius: '10px 0 10px 10px',
+		backgroundColor: '#f4f6fa',
+		backgroundImage: 'none',
+		color: '#91a3c0',
+		fontWeight: 600,
+		borderRadius: '10px 10px 0 10px',
+		maxWidth: '60%'
+	},
+	inputRow: {
+		background: 'white',
+		position: 'fixed',
+		bottom: 0,
+		right: 0,
+		padding: 20,
+		left: 342
+		// [theme.breakpoints.up('xs')]: {}
+	},
+	form: {
+		width: '100%'
+	},
+	avatar: {
+		margin: theme.spacing(1, 1.5),
+		minWidth: 0
+	},
+	listItem: {
+		display: 'flex',
+		width: '100%'
+	},
+	listItemRight: {
+		flexDirection: 'row-reverse'
+	},
+	messageInput: {
+		height: 70,
+		borderRadius: 8,
+		backgroundColor: '#F4F6FA',
+		display: 'flex',
+		paddingLeft: 20,
+		alignItems: 'center'
+	}
+}));
+
+const ChatBox = (props) => {
+	const [ currentUserId, setCurrentUserId ] = useState(null);
+	const [ newMessage, setNewMessage ] = useState('');
+	const [ messages, setMessages ] = useState([]);
+	const [ lastMessage, setLastMessage ] = useState(null);
+	const verifyUser = useVerify();
+
+	const getConversationMessages = useGetConversationMessages();
+	const sendConversationMessage = useSendConversationMessage();
+
+	let chatBottom = useRef(null);
+	const classes = useStyles();
+
+	useEffect(
+		() => {
+			reloadMessages();
+			scrollToBottom();
+		},
+		[ lastMessage, props.scope, props.conversationId ]
+	);
+
+	useEffect(() => {
+		const socket = socketIOClient(process.env.REACT_APP_API_URL, {
+			transports: [ 'websocket', 'polling', 'flashsocket' ]
+		});
```

Comment:
We shouldn't actually be create a new socket connection for every component, especially inside the chatbox component. We will need to have the socket object accessible across the whole application

Reply:
okay I moved this into the main Chat.jsx